ironfish-cli/src/commands/airdrop/raw-transactions.ts

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { isValidPublicAddress } from '@ironfish/rust-nodejs'
+import { Flags } from '@oclif/core'
+import fs from 'fs/promises'
+import { IronfishCommand } from '../../command'
+import { LocalFlags } from '../../flags'
+import { parseAllocationsFile } from '../../utils/allocations'
+import { AIRDROP_NOTES_IN_BLOCK, FEE_ORE_PER_AIRDROP } from './constants'
+
+export default class AirdropRawTransactions extends IronfishCommand {
+  static aliases = ['airdrop:raw']
+  static hidden = true
+
+  static flags = {
+    ...LocalFlags,
+    account: Flags.string({
+      required: true,
+      description: 'The name of the account used to create raw transactions',
+    }),
+    allocations: Flags.string({
+      required: true,
+      description:
+        'A CSV file with the format address,amountInIron,memo containing genesis block allocations',
+    }),
+    raw: Flags.string({
+      required: false,
+      default: 'raw_transactions.txt',
+      description: 'where to output the raw transactions',
+    }),
+  }
+  async start(): Promise<void> {
+    const { flags } = await this.parse(AirdropRawTransactions)
+    const account = flags.account
+    const client = await this.sdk.connectRpc()
+
+    const csv = await fs.readFile(flags.allocations, 'utf-8')
+    const result = parseAllocationsFile(csv)
+
+    if (!result.ok) {
+      this.error(result.error)
+    }
+    const allocations = result.allocations
+
+    await fs.rm(flags.raw)
+    const fileHandle = await fs.open(flags.raw, 'a')
+    for (let i = 0; i < allocations.length; i += AIRDROP_NOTES_IN_BLOCK) {
+      const chunk = allocations.slice(i, i + AIRDROP_NOTES_IN_BLOCK)
+      const outputs = []
+      for (const output of chunk) {
+        if (!isValidPublicAddress(output.publicAddress)) {
+          this.warn(
+            `Invalid public address ${output.publicAddress} for user: ${output.memo}, skipping`,
+          )
+          continue
+        }
+        outputs.push({
+          publicAddress: output.publicAddress,
+          amount: output.amountInOre.toString(),
+          memo: output.memo,
+        })
+      }
+
+      const result = await client.createTransaction({
+        account,
+        outputs,
+        fee: String(BigInt(AIRDROP_NOTES_IN_BLOCK) * FEE_ORE_PER_AIRDROP),
+      })
```

Comment:
I don't think `createTransaction` marks any notes as spent, so will this end up generating multiple transactions that spend the same note, if the note is large enough?

Reply:
Oh interesting, how should we mark the note spent? Or should we use a different method call here?

Reply:
The wallet will only mark notes as spent when a posted transaction is added. 

That won't work for us here because we can't alternate create/post 🤔

Reply:
Adding the ability to spend specific notes would be nice, but might be outside the scope of this PR.

Another option could be to boot up a node in-process (rather than connecting to one over RPC), then you can reach into the wallet directly and call a function on it. You could just call this and I think it would be good enough to prevent the note from appearing in getUnspentNotes:

```ts
await this.walletDb.deleteUnspentNoteHash(this, spentNoteHash, spentNote, tx)
```

Or make a custom `createTransaction` that takes a note to spend. The tricky bit will be generating the correct witness for the note.

Reply:
https://iron-fish.slack.com/archives/C02PF56HESK/p1681337839023619

https://github.com/iron-fish/ironfish/pull/3779

Reply:
Note that I decided to make this a separate RPC because I want to run this code in the simulator. I don't think the simulator is set up well to handle new node spawns in CLI layer, so I just forked cli command and added these lines as the only changes (thanks @hughy for help converting note types ):
```
      for (const spend of transaction.spends) {
        const decryptedNote = await account.getDecryptedNote(spend.note.hash())
        Assert.isNotUndefined(decryptedNote, 'Decrypted note that is being deleted is null.')
        await node.wallet.walletDb.deleteUnspentNoteHash(
          account,
          spend.note.hash(),
          decryptedNote,
        )
      }
```